### PR TITLE
Migration of test_glusterd_selinux

### PR DESCRIPTION
**Pull request by nishith-vihar**
[test_glusterd_selinux.py](https://github.com/gluster/glusto-tests/blob/master/tests/functional/glusterd/test_glusterd_selinux.py) test case is migrated to redant

Updates: #292
Fixes: #334 
Signed-off-by: Nishith Vihar Sakinala nsakinal@redhat.com

### CODE REVIEW

File: tests/functional/glusterd/test_glusterd_selinux.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                                    f"'{fqpath}' in {server} ")
+                continue
+
+            # Check owner of xml file
+            command = self.get_cmd('rpm', 'qf', fqpath)
+            ret = redant.execute_io_cmd(command, server)
+            exp_str = 'glusterfs-server'
+            ret_msg = " ".join(ret['msg'])
+            if exp_str not in ret_msg:
+                redant.logger.error(f"Fail: Owner of {fqpath} should be "
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_glusterd_selinux.py:58]: Same as above

File: tests/functional/glusterd/test_glusterd_selinux.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                                    f"{exp_str} on {server}")
+
+            # Validate SELinux label
+            command = self.get_cmd('ls', 'lZ', fqpath)
+            ret = redant.execute_io_cmd(command, server)
+            exp_str = 'system_u:object_r:lib_t:s0'
+            if exp_str not in ret_msg:
+                redant.logger.error(f"Fail: SELinux label on {fqpath}"
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_glusterd_selinux.py:66]: Same as above

**Review by nik-redhat — CHANGES_REQUESTED** [review]

File: tests/functional/glusterd/test_glusterd_selinux.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        2. Validate the owner of this file as 'glusterfs-server'
+        3. Validate SELinux label context as 'system_u:object_r:lib_t:s0'
+        """
+
+        fqpath = '/usr/lib/firewalld/services/glusterfs.xml'
+
+        for server in self.server_list:
+            # Check existence of xml file
+            if not redant.path_exists(server, fqpath):
+                redant.logger.error("Failed to verify existence of"
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_glusterd_selinux.py:48]: It should raise an exception instead.

**Review by nik-redhat — CHANGES_REQUESTED** [review]

File: tests/functional/glusterd/test_glusterd_selinux.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                                    f"'{fqpath}' in {server} ")
+                continue
+
+            # Check owner of xml file
+            command = self.get_cmd('rpm', 'qf', fqpath)
+            ret = redant.execute_io_cmd(command, server)
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_glusterd_selinux.py:54]: Update to use execute_abstract_op_node

File: tests/functional/glusterd/test_glusterd_selinux.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            exp_str = 'glusterfs-server'
+            ret_msg = " ".join(ret['msg'])
+            if exp_str not in ret_msg:
+                redant.logger.error(f"Fail: Owner of {fqpath} should be "
+                                    f"{exp_str} on {server}")
+
+            # Validate SELinux label
+            command = self.get_cmd('ls', 'lZ', fqpath)
+            ret = redant.execute_io_cmd(command, server)
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_glusterd_selinux.py:63]: Update to use execute_abstract_op_node

File: tests/functional/glusterd/test_glusterd_selinux.py (hunk 4 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_glusterd_selinux.py:54]: Ack

File: tests/functional/glusterd/test_glusterd_selinux.py (hunk 5 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_glusterd_selinux.py:63]: Ack

File: tests/functional/glusterd/test_glusterd_selinux.py (hunk 3 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_glusterd_selinux.py:48]: Ack

File: tests/functional/glusterd/test_glusterd_selinux.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License along
+  with this program; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+
+Description:
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_glusterd_selinux.py:19]: nit: indent as per above doc string.

Comment (nishith-vihar) [reply tests/functional/glusterd/test_glusterd_selinux.py:19]: Ack

**Review by nik-redhat — APPROVED** [review]

**Pull request merged**